hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -765,6 +774,7 @@ protected Object putInternal(Data key, Object value, long ttl, boolean loadFromS
         if (record == null) {
             record = createRecord(value, ttl, now);
             storage.put(key, record);
+            eventJournal.writeAddEvent(mapContainer.getObjectNamespace(), partitionId, record.getKey(), record.getValue());
```

Comment:
It's a matter of flavour - on the other hand I would not want to receive an event for something that failed to be stored in the `RecordStore`.
Currently the event journal cannot be configured with a ringbuffer store so there is no way it could store to a database. And the choice was that it should contain an event only if the call to the `RecordStore` succeeded.